### ISSUE TITLE
ERROR : type 'Null' is not a subtype of type 'Map<String, dynamic>' in type cast

### ISSUE THREAD

**Issue by Avadhesh-Pandey** [aissat/easy_localization#545]
Type case error coming up when running the app with null safety and fallback.

**Reason for the issue**
easy_localization_controller.dart class loadTranslations() method's below mentioned lines is trying to find the translation with just the languageCode. On failure, throwing type case error - type 'Null' is not a subtype of type 'Map<String, dynamic>' in type cast

```
if (_locale.countryCode != null && _locale.countryCode!.isNotEmpty) {
          baseLangData =
              await loadBaseLangTranslationData(Locale(locale.languageCode));
        }

```

**Map locales used** - {"en_ID": en_ID, "id_ID": id_ID, "bn_BD": bn_BD, "en_UK": en_UK}

Fallback map locale : "en_UK"

Find the details in Screen shot attached, wherein the locale in **bn_BD** but library tries to find the translations for "bn" and fails with and type case error as mentioned above.

<img width="785" alt="image" src="https://user-images.githubusercontent.com/42084941/211290423-c341d349-d77e-498a-b4c0-67098098311c.png">

Note : Its working fine when running without null safety or without fallback [locale.](url)

**Comment by definitelyme** [comment]
Any solution yet? @aissat @Avadhesh-Pandey @RomainFranceschini @3ace @xurei 
Translation doesn't work with Language code and Country code when declared in `EasyLocalization.startLocale`. Throws a runtime error (Flutter Red Screen), that is not printed to the console.

I'd be happy to open a PR for this?

![Screenshot_1677499938](https://user-images.githubusercontent.com/28307747/221560836-6920c10e-5573-45fe-88ad-f42dce978d73.png)

**Comment by TheMeanCanEHdian** [comment]
This error appears to be twofold:
1. [loadBaseLanTranslationData](https://github.com/aissat/easy_localization/blob/develop/lib/src/easy_localization_controller.dart#L123-L129) was added so that if you had locales such as `en_US` and `en_UK` it first looks for any translations that might exist under just `en`. This allows for common words between all English languages to be under the `en` locale to reduce duplicate translations (think the words stop and start).
However, loadBaseLanTranslationData calls [loadTranslationData](https://github.com/aissat/easy_localization/blob/develop/lib/src/easy_localization_controller.dart#L123-L129) and this function does not allow for null results to be returned. This is a problem since that common locale `en` may not exist and assetLoader would then be expected to return `null`.
2. The [load](https://github.com/aissat/easy_localization/blob/develop/lib/src/asset_loader.dart#L20) function under AssetLoader allows for a `null` value to be returned. However, the [load override](https://github.com/aissat/easy_localization/blob/develop/bin/generate.dart#L248-L250) for codegen_loader does not allow for a `null` value.

Both of these issues each contribute to the underlying problem.

I have created the PR #570 that contains fixes I have tested in my application [Tautulli Remote](https://github.com/Tautulli/Tautulli-Remote). While waiting for the PR to be merged you can use my forked branch in your pubspec by changing
```yaml
easy_localization: ^3.0.1
```
to 
```yaml
easy_localization:
    git: 
      url: https://github.com/TheMeanCanEHdian/easy_localization.git
      ref: nullTranslationFix
```

**Comment by bw-flagship** [comment]
Seems to be fixed by #570
